tests/ui/parser/attribute/attr-unquoted-ident.rs

```diff
@@ -0,0 +1,15 @@
+// compile-flags: -Zdeduplicate-diagnostics=yes
+// run-rustfix
+
+fn main() {
+    #[cfg(key=foo)]
+    //~^ ERROR expected unsuffixed literal or identifier, found `foo`
```

Comment:
this error message is still confusing, `foo` *is* an identifier. can you change the error so that it mentions an identifier only of is really allowed?

Reply:
I changed the error so the "or identifier" part only gets shown when we don't try to suggest quoting the ident.